app/scripts/controllers/transactions/index.js

```diff
@@ -140,6 +140,10 @@ const METRICS_STATUS_FAILED = 'failed on-chain';
  * @param {number} [opts.txHistoryLimit] - number *optional* for limiting how many transactions are in state
  * @param {Function} opts.hasCompletedOnboarding - Returns whether or not the user has completed the onboarding flow
  * @param {object} opts.preferencesStore
+ * @param {object} hooks
+ * @param {Function} hooks.afterSign - Determines whether to execute logic after signing a transaction.
```

Comment:
Maybe:

`Additional logic to execute after signing a transaction. Return false to not change the status to signed.`

We could use a similar format for the other descriptions too?